File: TrabajoAcademicoNet/SoftProgPersistencia/DAOImpl/DAOImplPersona/AdministradorSistemaDAOImpl.cs
using PUCP.Inf30.SoftProg.Modelo.PagoModelo;
using PUCP.Inf30.SoftProg.Modelo.PagoModelo.PagoUtils;
using PUCP.Inf30.SoftProg.Modelo.PersonaModelo;
using PUCP.Inf30.SoftProg.Modelo.PersonaModelo.PersonaUtils;
using PUCP.Inf30.SoftProg.Persistencia.DAO.DAOPersona;
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Common;
using System.Data.Odbc;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PUCP.Inf30.SoftProg.Persistencia.DAOImpl.DAOImplPersona
{
    public class AdministradorSistemaDAOImpl :
        BaseDAOImpl<AdministradorSistema>, IAdministradorSistemaDAO
    {
        protected override DbCommand ComandoActualizar(DbConnection conn, AdministradorSistema administradorSistema)
        {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "modificarAdministrador";
            cmd.CommandType = CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_idAdministrador", DbType.Int32, administradorSistema.Id);
            this.AgregarParametroEntrada(cmd, "@p_cargo", DbType.String, administradorSistema.Cargo);
            this.AgregarParametroEntrada(cmd, "@p_rango", DbType.String, administradorSistema.Rango);
            this.AgregarParametroEntrada(cmd, "@p_dni", DbType.String, administradorSistema.Dni);
            this.AgregarParametroEntrada(cmd, "@p_nombre", DbType.String, administradorSistema.Nombre);
            this.AgregarParametroEntrada(cmd, "@p_apellidoPaterno", DbType.String, administradorSistema.ApellidoPaterno);
            this.AgregarParametroEntrada(cmd, "@p_apellidoMaterno", DbType.String, administradorSistema.ApellidoMaterno);
            this.AgregarParametroEntrada(cmd, "@p_genero", DbType.String, administradorSistema.Genero);
            this.AgregarParametroEntrada(cmd, "@p_fechaNacimiento", DbType.DateTime, administradorSistema.FechaNacimiento);
            this.AgregarParametroEntrada(cmd, "@p_telefono", DbType.Int32, administradorSistema.Telefono);
            this.AgregarParametroEntrada(cmd, "@p_Sueldo", DbType.Double, administradorSistema.Sueldo);
            this.AgregarParametroEntrada(cmd, "@p_Activo", DbType.Boolean, administradorSistema.Activo);
            
            return cmd;
        }

        protected override DbCommand ComandoCrear(DbConnection conn, AdministradorSistema administradorSistema)
        {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "insertarAdministrador";
            cmd.CommandType = CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_idAdministrador", DbType.Int32, administradorSistema.Id);
            this.AgregarParametroEntrada(cmd, "@p_cargo", DbType.String, administradorSistema.Cargo);
            this.AgregarParametroEntrada(cmd, "@p_rango", DbType.String, administradorSistema.Rango);
            this.AgregarParametroEntrada(cmd, "@p_dni", DbType.String, administradorSistema.Dni);
            this.AgregarParametroEntrada(cmd, "@p_nombre", DbType.String, administradorSistema.Nombre);
            this.AgregarParametroEntrada(cmd, "@p_apellidoPaterno", DbType.String, administradorSistema.ApellidoPaterno);
            this.AgregarParametroEntrada(cmd, "@p_apellidoMaterno", DbType.String, administradorSistema.ApellidoMaterno);
            this.AgregarParametroEntrada(cmd, "@p_genero", DbType.String, administradorSistema.Genero);
            this.AgregarParametroEntrada(cmd, "@p_fechaNacimiento", DbType.DateTime, administradorSistema.FechaNacimiento);
            this.AgregarParametroEntrada(cmd, "@p_telefono", DbType.Int32, administradorSistema.Telefono);
            this.AgregarParametroEntrada(cmd, "@p_Sueldo", DbType.Double, administradorSistema.Sueldo);
            this.AgregarParametroEntrada(cmd, "@p_Activo", DbType.Boolean, administradorSistema.Activo);

            return cmd;
        }

        protected override DbCommand ComandoEliminar(DbConnection conn, int id)
        {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "eliminarAdministrador";
            cmd.CommandType = CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_idAdministrador", DbType.Int32, id);

            return cmd;
        }

        protected override DbCommand ComandoLeer(DbConnection conn, int id)
        {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "buscarAdministradorPorId";
            cmd.CommandType = CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_idAdministrador", DbType.Int32, id);

            return cmd;
        }

        protected override DbCommand ComandoLeerTodos(DbConnection conn)
        {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "leerAdministradores";
            cmd.CommandType = CommandType.StoredProcedure;
            return cmd;
        }

        protected override AdministradorSistema MapearModelo(DbDataReader reader)
        {
            return new AdministradorSistema
            {
                Id = Convert.ToInt32(reader["idLineaComprobantePago"]),
                Rango = (Jerarquia)Enum.Parse(typeof(Jerarquia), Convert.ToString(reader["rango"])),
                Cargo = (Cargo)Enum.Parse(typeof(Cargo), Convert.ToString(reader["cargo"])),
                Dni = Convert.ToString(reader["dni"]),
                Nombre = Convert.ToString(reader["nombre"]),
                ApellidoPaterno = Convert.ToString(reader["apellidoPaterno"]),
                ApellidoMaterno = Convert.ToString(reader["apellidoMaterno"]),
                Genero = (Genero)Enum.Parse(typeof(Genero), Convert.ToString(reader["genero"])),
                FechaNacimiento = Convert.ToDateTime(reader["fechaNacimiento"]),
                Telefono = Convert.ToInt32(reader["telefono"]),
                Sueldo = Convert.ToDouble(reader["sueldo"]),
                Activo = Convert.ToBoolean(reader["activo"])

            };
        }
    }
}

File: Java/TrabajoAcademico/SoftProgDAO/src/main/java/pe/edu/pucp/inf30/softprog/daoimpl/persona/AdministradorSistemaDAOImpl.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.edu.pucp.inf30.softprog.daoimpl.persona;

import java.sql.CallableStatement;
import java.sql.Connection;
import java.sql.Date;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Types;
import pe.edu.pucp.inf30.softprog.dao.persona.AdministradorSistemaDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.BaseDAO;
import pe.edu.pucp.inf30.softprog.modelo.persona.AdministradorSistema;

/**
 *
 * @author Cristhian Horacio
 */
public class AdministradorSistemaDAOImpl extends BaseDAO<AdministradorSistema> implements AdministradorSistemaDAO {

    @Override
    protected PreparedStatement comandoCrear(Connection conn, AdministradorSistema modelo) throws SQLException {
        String sql = "{CALL insertarAdministrador(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)}";
        
        CallableStatement cmd = conn.prepareCall(sql);
        
        cmd.setString("p_cargo", modelo.getCargoString());
        cmd.setString("p_rango", modelo.getJerarquiaString());
        cmd.setString("p_dni", modelo.getDni());
        cmd.setString("p_nombre", modelo.getNombre());
        cmd.setString("p_apellidoPaterno", modelo.getApellidoPaterno());
        cmd.setString("p_apellidoMaterno", modelo.getApellidoMaterno());
        cmd.setString("p_genero", modelo.getGeneroString());
        cmd.setDate("p_fechaNacimiento", java.sql.Date.valueOf(modelo.getFechaNacimiento()));
        cmd.setInt("p_telefono", modelo.getTelefono());
        cmd.setDouble("p_Sueldo", modelo.getSueldo());
        cmd.setInt("p_Activo", modelo.getActivoInt());
        cmd.setInt("p_idCuentaUsuario", modelo.getCuenta().getId());
        cmd.registerOutParameter("p_id", Types.INTEGER);
        
        return cmd;
    }

    @Override
    protected PreparedStatement comandoActualizar(Connection conn, AdministradorSistema modelo) throws SQLException {
        String sql = "{CALL modificarAdministrador(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)}";
        
        CallableStatement cmd = conn.prepareCall(sql);
        
        cmd.setInt("p_idAdministrador", modelo.getId());
        cmd.setString("p_cargo", modelo.getCargoString());
        cmd.setString("p_rango", modelo.getJerarquiaString());
        cmd.setString("p_dni", modelo.getDni());
        cmd.setString("p_nombre", modelo.getNombre());
        cmd.setString("p_apellidoPaterno", modelo.getApellidoPaterno());
        cmd.setString("p_apellidoMaterno", modelo.getApellidoMaterno());
        cmd.setString("p_genero", modelo.getGeneroString());
        cmd.setDate("p_fechaNacimiento", java.sql.Date.valueOf(modelo.getFechaNacimiento()));
        cmd.setInt("p_telefono", modelo.getTelefono());
        cmd.setDouble("p_Sueldo", modelo.getSueldo());
        cmd.setInt("p_Activo", modelo.getActivoInt());
        cmd.setInt("p_idCuentaUsuario", modelo.getCuenta().getId());
        
        return cmd;
    }

    @Override
    protected PreparedStatement comandoEliminar(Connection conn, Integer id) throws SQLException {
        String sql = "{CALL eliminarAdministrador(?)}";
        
        CallableStatement cmd = conn.prepareCall(sql);
        
        cmd.setInt("p_idAdministrador", id);
        
        return cmd;
    }

    @Override
    protected PreparedStatement comandoLeer(Connection conn, Integer id) throws SQLException {
        String sql = "{CALL buscarAdministradorPorId(?)}";
        
        CallableStatement cmd = conn.prepareCall(sql);
        
        cmd.setInt("p_idAdministrador", id);
        
        return cmd;
    }

    @Override
    protected PreparedStatement comandoLeerTodos(Connection conn) throws SQLException {
        String sql = "{CALL listarAdministradores()}";
        
        CallableStatement cmd = conn.prepareCall(sql);
        return cmd;
    }

    @Override
    protected AdministradorSistema mapearModelo(ResultSet rs) throws SQLException {
        AdministradorSistema administrador = new AdministradorSistema();
        
        administrador.setId(rs.getInt("idAdministrador"));
        administrador.setCargoString(rs.getString("cargo"));
        administrador.setJerarquiaString(rs.getString("jerarquia"));
        administrador.setDni(rs.getString("dni"));
        administrador.setNombre(rs.getString("nombre"));
        administrador.setApellidoPaterno(rs.getString("apellidoPaterno"));
        administrador.setApellidoMaterno(rs.getString("apellidoMaterno"));
        administrador.setGeneroString(rs.getString("genero"));
        administrador.setFechaNacimiento(rs.getDate("fechaNacimiento").toLocalDate());
        administrador.setTelefono(rs.getInt("telefono"));
        administrador.setSueldo(rs.getDouble("Sueldo"));
        administrador.setActivoInt(rs.getInt("activo"));
        administrador.setCuenta(new CuentaUsuarioDAOImpl().leer(rs.getInt("idCuentaUsuario")));
        
        return administrador;
    }
    
}
